resp/get_foxsi_optics_effarea.pro

```diff
@@ -1,73 +1,138 @@
+;			      PLOT - plot to the current device
+;			      OFFAXIS_ANGLE - off-axis angle. if array then [pan, tilt] in arcmin
+;			      YEAR - year of the FOXSI launch. It can be: 2012, 2014 or 2018. If not set, COMMON DATE will be used instead.
 ;
 ;WRITTEN: Steven Christe (21-Jan-15)
 ;	modified:	LG	2015 Feb	Switched X0->D6 and X6->D0
 ;	modified:	SM	2019 Dec	Switched to optic module numbers instead of detector position
+; modified: MB  2020 Jan  Differenciate among the three foxsi flights using the DATE keyword.
+;               2020 Feb  Add a YEAR keyword that can be used when the user don't want to load all FOXSI COMMON variables.
 
 default, data_dir, 'calibration_data/'
 default, offaxis_angle, [0.0, 0.0]
 default, module_number, 0
 
 IF n_elements(offaxis_angle) EQ 1 THEN angle = 1/sqrt(2) * [offaxis_angle, offaxis_angle] $
-    ELSE angle = offaxis_angle
+ELSE angle = offaxis_angle
+
+IF KEYWORD_SET(YEAR) THEN BEGIN
+  IF ((YEAR EQ 2012) OR (YEAR EQ 2014) OR (YEAR EQ 2018)) THEN BEGIN
+    ; foxsi1 :
+    IF (YEAR EQ 2012) THEN BEGIN
+      IF (WHERE(MODULE_NUMBER EQ [0,1,2,3,4,5,6]) NE -1) THEN BEGIN
+        files =  GETENV('FOXSIPKG') + '/' + data_dir + 'FOXSI2_' + ['Module_X-' + num2str(MODULE_NUMBER) + '_EA_pan.txt', $
+                 'Module_X-' + num2str(MODULE_NUMBER) + '_EA_tilt.txt'] ; for foxsi1 we used the same EA files as for foxsi2.
+      ENDIF ELSE PRINT, 'Invalid Module_number for FOXSI1. Chose one of 0,1,2,3,4,5 or 6.'
+    ENDIF
+    ; foxsi2 :
+    IF YEAR EQ 2014 THEN BEGIN
+      IF (WHERE(MODULE_NUMBER EQ [0,1,2,3,4,5,6]) NE -1) THEN BEGIN
+        files =  GETENV('FOXSIPKG') + '/' + data_dir + 'FOXSI2_' + ['Module_X-' + num2str(MODULE_NUMBER) + '_EA_pan.txt', $
+                 'Module_X-' + num2str(MODULE_NUMBER) + '_EA_tilt.txt']
+      ENDIF ELSE PRINT, 'Invalid Module_number for FOXSI2. Chose one of 0,1,2,3,4,5 or 6.'
+    ENDIF
+    ; foxsi3 :
+    IF YEAR EQ 2018 THEN BEGIN
+      IF (WHERE(MODULE_NUMBER EQ [0,1,2,4,5]) NE -1) THEN BEGIN
+        files =  GETENV('FOXSIPKG') + '/' + data_dir + 'FOXSI2_' + ['Module_X-' + num2str(MODULE_NUMBER) + '_EA_pan.txt', $
+                 'Module_X-' + num2str(MODULE_NUMBER) + '_EA_tilt.txt']
+      ENDIF ELSE BEGIN
+        IF (WHERE(MODULE_NUMBER EQ [7,8]) NE -1) THEN BEGIN
+          files =  GETENV('FOXSIPKG') + '/' + data_dir + 'FOXSI3_' + ['Module_X-' + num2str(MODULE_NUMBER) + '_EA_pan.txt', $
+                   'Module_X-' + num2str(MODULE_NUMBER) + '_EA_tilt.txt']
+        ENDIF ELSE PRINT, 'Invalid Module_number for FOXSI3. Chose one of 0,1,2,4,5,7 or 8.'
+      ENDELSE
+    ENDIF
+  ENDIF ELSE BEGIN
+    PRINT, 'YEAR has an illegal value.'
+  ENDELSE
+ENDIF ELSE BEGIN
+  ; If YEAR is not set then COMMON DATE will be used to know wich FOXSI flight the user is working with. 
+  IF FOXSI_PARAM EQ !NULL THEN BEGIN
+    print,'Please initialize the FOXSI common parameter or provide the year of the flight via the YEAR keyword'
```

Comment:
I would add a line after printing the error message (within the IF statement) of "return, -1" so that you leave the function when this error occurs.

Reply:
Good peace of advice! Implemented already :) https://github.com/foxsi/foxsi-science/pull/85/commits/f94fbb402877420960da95f77526ffb1e6dc19c0